File: VideoEidtexter/src/main/java/com/cgfay/cameralibrary/bean/ItemSticker.java
package com.cgfay.cameralibrary.bean;

import android.os.Environment;

import java.util.ArrayList;
import java.util.List;

public class ItemSticker {
    private String name;
    private String path;
    public static int TYPE_GIF = 1;
    public static int TYPE_PNG = 0;
    private int type = TYPE_PNG;

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public static List<ItemSticker> getStickerList() {
        String path = Environment.getExternalStorageDirectory().getAbsolutePath() + "/Download/";
        List<ItemSticker> list = new ArrayList<>();
        ItemSticker itemSticker = new ItemSticker();
        itemSticker.setName("sticker");
        itemSticker.setPath(path + "sticker_02.png");
        list.add(itemSticker);

        ItemSticker itemSticker1 = new ItemSticker();
        itemSticker1.setName("sticker1");
        itemSticker1.setPath(path + "sticker_05.png");
        list.add(itemSticker1);

        ItemSticker itemSticker2 = new ItemSticker();
        itemSticker2.setName("sticker2");
        itemSticker2.setPath(path + "sticker_13.png");
        list.add(itemSticker2);


        ItemSticker itemSticker3 = new ItemSticker();
        itemSticker3.setName("sticker3");
        itemSticker3.setPath(path + "test1.gif");
        itemSticker3.type = itemSticker3.TYPE_GIF;

        list.add(itemSticker3);

        ItemSticker itemSticker4 = new ItemSticker();
        itemSticker4.setName("sticker4");
        itemSticker4.setPath(path + "test2.gif");
        itemSticker4.type = itemSticker4.TYPE_GIF;
        list.add(itemSticker4);


        return list;
    }
}

File: videov2/src/main/java/com/owoh/video/ItemSticker.kt
package com.owoh.video

import android.os.Environment
import java.util.*

class ItemSticker {
    val TYPE_GIF = 1
    val TYPE_PNG = 0
    var name: String? = null
    var path: String? = null
    var type: Int = TYPE_PNG;
    val TYPE_TEXT =2
    companion object {

        val stickerList: List<ItemSticker>
            get() {
                val path = Environment.getExternalStorageDirectory().absolutePath + "/Download/"
                val list = ArrayList<ItemSticker>()
                val itemSticker = ItemSticker()
                itemSticker.name = "sticker"
                itemSticker.path = path + "test1.jpg"
                itemSticker.type = itemSticker.TYPE_PNG
                list.add(itemSticker)

                val itemSticker1 = ItemSticker()
                itemSticker1.name = "sticker1"
                itemSticker1.path = path + "test1.jpg"
                itemSticker1.type = itemSticker1.TYPE_PNG
                list.add(itemSticker1)

                val itemSticker2 = ItemSticker()
                itemSticker2.name = "sticker2"
                itemSticker2.path = path + "test1.jpg"
                itemSticker2.type = itemSticker2.TYPE_PNG
                list.add(itemSticker2)

                val itemSticker3 = ItemSticker()
                itemSticker3.name = "sticker3"
                itemSticker3.path = path + "test2.gif"
                itemSticker3.type = itemSticker3.TYPE_GIF
                list.add(itemSticker3)

                val itemSticker4 = ItemSticker()
                itemSticker4.name = "sticker4"
                itemSticker4.path = path + "test3.gif"
                itemSticker4.type = itemSticker4.TYPE_GIF
                list.add(itemSticker4)

                val itemSticker6 = ItemSticker()
                itemSticker6.name = "sticker4"
                itemSticker6.path = path + "watermark.gif"
                itemSticker6.type = itemSticker6.TYPE_GIF
                list.add(itemSticker6)

                val textSticker = ItemSticker()
                textSticker.name = "文字"
                textSticker.path = path + "test1.jpg"
                textSticker.type = textSticker.TYPE_TEXT
                list.add(textSticker)
                return list
            }
    }
}
